packages/firebase_dart/lib/src/database/impl/treestructureddata/treestructureddata.dart

```diff
@@ -447,3 +452,20 @@ class UnmodifiableFilteredMap<K extends Comparable, V>
         start: start, end: end, limit: limit, reversed: reversed);
   }
 }
+
+extension on Iterable<Name> {
+  bool get isList {
+    try {
+      final keys = [
+        // Bail out early if all keys are not integers
+        for (final n in this) n.asInt() ?? (throw Exception('NotList'))
+      ].sorted((a, b) => a.compareTo(b));
```

Comment:
Can we assume that the keys are already sorted or not? That would speed this up a bit. Though if it is already sorted, this will be relatively fast anyways.